internal/chmod_sock.go

```diff
@@ -0,0 +1,17 @@
+package internal
+
+import (
+	"log"
+	"os"
+)
+
+// UpdateSocketFilePermissions set socket file permissions
+func UpdateSocketFilePermissions(sockFileName string) {
+	if !FileExists(sockFileName) {
+		return
+	}
+	// #nosec G302 -- need world writable permissions
+	if err := os.Chmod(sockFileName, PermUserRWGroupRWOthersRW); err != nil {
```

Comment:
Under snap environment, in case of error changing permissions, the client will not be able to communicate with daemon - we will log error into log file and that's it, nothing more we can do about it.